test/e2e/k8petstore.go

```diff
@@ -0,0 +1,170 @@
+
+	cmd := exec.Command(
+		k8bpsScriptLocation,
+		testContext.KubectlPath,
+		k8bpsContainerVersion,
+		k8bpsThroughputDummy,
+		strconv.Itoa(restServers),
+		strconv.Itoa(loadGenerators),
+		k8bpsRedisSlaves,
+		k8bpsDontRunTest, // Don't bother embedded test.
+		ns,
+	)
+
+	cmd.Stdout = os.Stdout
+	cmd.Stderr = os.Stderr
+
+	Logf("Starting k8petstore application....")
+	// Run the k8petstore app, and log / fail if it returns any errors.
+	// This should return quickly, assuming containers are downloaded.
+	if err = cmd.Start(); err != nil {
+		log.Fatal(err)
+	}
+	//Make sure there are no command errors.
+	if err = cmd.Wait(); err != nil {
+		if exiterr, ok := err.(*exec.ExitError); ok {
+			if status, ok := exiterr.Sys().(syscall.WaitStatus); ok {
+				log.Printf("Exit Status: %d", status.ExitStatus())
+			}
+		}
+	}
+	Expect(err).NotTo(HaveOccurred())
+	Logf("... Done starting k8petstore ")
+
+	totalTransactions := 0
+	Logf("Start polling, timeout is %v seconds", maxTime)
+
+	//How long until the FIRST transactions are created.
+	startupTimeout := time.After(time.Duration(k8bpsStartupTimeout))
+
+	//Maximum time to wait until we reach the nth transaction.
+	transactionsCompleteTimeout := time.After(time.Duration(maxTime))
+	tick := time.Tick(2 * time.Second)
+	var ready = false
+
+	Logf("Now waiting %v seconds to see progress (transactions  > 3)", k8bpsStartupTimeout)
+T:
+	for {
+		select {
+		case <-transactionsCompleteTimeout:
+			Logf("Timeout %v reached, transactions not complete.  Breaking!", tick)
+			break T
+		case <-startupTimeout:
+			err, totalTransactions = readTransactions(c, ns)
+			Logf("Timeout %v reached.  Checking if transactions have occured.", startupTimeout)
+			//If we don't have 3 transactions, fail the test.
+			if err != nil {
+				Logf("Failed : Error %v", err)
+				break T
+			}
+			if totalTransactions < 3 {
+				break T
+			}
+			ready = true
+		case <-tick:
+			//Pass if we've collected enough transactions
+			err, totalTransactions = readTransactions(c, ns)
+			if ready {
+				Expect(err).NotTo(HaveOccurred())
+			}
+			if totalTransactions > minExpected {
+				break T
+			}
+			Logf("Time: %v, %v = total petstore transactions stored into redis.", time.Now(), totalTransactions)
+		}
+	}
+
+	//We should have exceeded the minExpected num of transactions.
+	//If this fails, but there are transactions being created, we may need to recalibrate
+	//the minExpected value - or else - your cluster is broken/slow !
+	Ω(totalTransactions).Should(BeNumerically(">", minExpected))
+}
+
+var _ = Describe("Pet Store", func() {
+
+	//The number of minions dictates total number of generators/transaction expectations.
+	var minionCount int
+	f := NewFramework("petstore")
+
+	It(fmt.Sprintf("should scale to persist a nominal number ( %v ) of transactions in %v seconds", k8bpsSmokeTestTransactions, k8bpsSmokeTestTimeout), func() {
```

Comment:
Can you please add "[Skipped][Example]" to the name of the test (similarly to what we have e.g. here:
https://github.com/kubernetes/kubernetes/blob/master/test/e2e/examples.go#L250